rskj-core/src/test/java/co/rsk/peg/feeperkb/FeePerKbIntegrationTest.java

```diff
@@ -0,0 +1,475 @@
+
+        // Again authorized callers vote
+
+        Coin seventhVoteFeePerKb  = Coin.valueOf(127_000L);
+        // First vote for a new value
+        voteFeePerKb(seventhVoteFeePerKb, 0);
+        //It still has the previous voting round value
+        assertFeePerKbValue(fourteenthFeePerKbVote);
+
+        Coin eighthVoteFeePerKb  = Coin.valueOf(128_000L);
+        // Second vote for a different value, different authorizer
+        voteFeePerKb(eighthVoteFeePerKb, 1);
+        //It still has the previous voting round value
+        assertFeePerKbValue(fourteenthFeePerKbVote);
+
+        Coin ninthVoteFeePerKb  = Coin.valueOf(129_000L);
+        // Third vote for a different value, different authorizer
+        voteFeePerKb(ninthVoteFeePerKb, 2);
+        //It still has the previous voting round value
+        assertFeePerKbValue(fourteenthFeePerKbVote);
+
+        // Again authorized callers vote
+
+        // First vote for a new value
+        voteFeePerKb(fifteenthFeePerKbVote, 0);
+        //It still has the previous voting round value
+        assertFeePerKbValue(fourteenthFeePerKbVote);
+
+        // Second vote for a same value than the previous one
+        voteFeePerKb(fifteenthFeePerKbVote, 1);
+        //It still has the previous voting round value
+        assertFeePerKbValue(fifteenthFeePerKbVote);
+    }
+
+    private void assertFeePerKbValue(Coin feePerKbExpectedResult) {
+        Coin feePerKbActualResult = feePerKbSupport.getFeePerKb();
+        assertEquals(feePerKbExpectedResult, feePerKbActualResult);
+    }
+
+    private void voteFeePerKb(Coin feePerKb, int authorizerIndex) {
+        RskAddress authorizerAddress = this.getAuthorizedRskAddresses().get(authorizerIndex);
+        Transaction voteTx = getTransactionFromCaller(authorizerAddress);
+        feePerKbSupport.voteFeePerKbChange(voteTx, feePerKb, signatureCache);
+    }
+
+    private void voteFeePerKbByUnauthorizedCaller(Coin feePerKb) {
+        RskAddress authorizerAddress = this.getUnauthorizedRskAddress();
+        Transaction voteTx = getTransactionFromCaller(authorizerAddress);
+        feePerKbSupport.voteFeePerKbChange(voteTx, feePerKb, signatureCache);
+    }
+
+    private Transaction getTransactionFromCaller(RskAddress authorizer) {
+        Transaction txFromAuthorizedCaller = mock(Transaction.class);
+        when(txFromAuthorizedCaller.getSender(signatureCache)).thenReturn(authorizer);
+
+        return txFromAuthorizedCaller;
+    }
+
+    private List<RskAddress> getAuthorizedRskAddresses() {
+        return Stream.of(
+            "a02db0ed94a5894bc6f9079bb9a2d93ada1917f3",
+            "180a7edda4e640ea5a3e495e17a1efad260c39e9",
+            "8418edc8fea47183116b4c8cd6a12e51a7e169c1"
+        ).map(RskAddress::new).collect(Collectors.toList());
+    }
+
+    private RskAddress getUnauthorizedRskAddress() {
+        return new RskAddress("e2a5070b4e2cb77fe22dff05d9dcdc4d3eaa6ead");
+    }
+}
```

Comment:
What happens if, for example, the current fee per kb is 100_000L and the authorizers decide to vote for the same 100_000L? Does it get rejected? Does it throw an error?

Reply:
Hey, Jere!
Brilliant contribution! In this case, 100_000L is set again as "the new fee per KB value".
I just added this case to the tests:
`16th voting: authorized callers vote the same current fee per KB value`